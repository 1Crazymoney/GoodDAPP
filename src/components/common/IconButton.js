// @flow
import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { Icon } from 'react-native-elements'

type IconProps = {
  text: String,
  onPress: Function,
  disabled: Boolean,
  name: String
}

/**
 * Returns a button with an icon and text
 *
 * @param {IconProps} props
 * @returns Button with icon and text
 */
const IconButton = ({ text, onPress, disabled, ...iconProps }: IconProps) => {
  const styles = createStyles(disabled)
  return (
    <View style={styles.container} onClick={!disabled ? onPress : undefined}>
      <Icon reverse color="white" reverseColor={disabled ? 'rgba(0, 0, 0, 0.32)' : '#282c34'} {...iconProps} />
      <Text style={styles.text}>{text}</Text>
    </View>
  )
}

const createStyles = disabled =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      flex: 1,
      cursor: disabled ? 'inherit' : 'pointer'
    },
    text: { color: disabled ? 'rgba(0, 0, 0, 0.32)' : 'inherit' }
  })

export default IconButton
