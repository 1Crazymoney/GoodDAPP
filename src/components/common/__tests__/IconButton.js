import React from 'react'
import IconButton from '../IconButton'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('IconButton enabled', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<IconButton text="edit" name="test" />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<IconButton text="edit" name="test" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('IconButton disabled', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<IconButton text="edit" name="test" disabled />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<IconButton text="edit" name="test" disabled />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
