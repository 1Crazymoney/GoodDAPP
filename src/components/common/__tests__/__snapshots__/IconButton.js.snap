// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton disabled matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxFlex": 1,
      "WebkitBoxOrient": "horizontal",
      "WebkitFlexBasis": "0%",
      "WebkitFlexDirection": "row",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "alignItems": "center",
      "cursor": "inherit",
      "flexBasis": "0%",
      "flexDirection": "row",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexAlign": "center",
      "msFlexDirection": "row",
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxPack": "center",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,1.00)",
        "borderBottomLeftRadius": "28px",
        "borderBottomRightRadius": "28px",
        "borderTopLeftRadius": "28px",
        "borderTopRightRadius": "28px",
        "height": "52px",
        "justifyContent": "center",
        "marginBottom": "7px",
        "marginLeft": "7px",
        "marginRight": "7px",
        "marginTop": "7px",
        "msFlexAlign": "center",
        "msFlexPack": "center",
        "width": "52px",
      }
    }
  >
    <div
      className="css-text-76zvg2"
      dir="auto"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.00)",
          "color": "rgba(0,0,0,0.32)",
          "fontFamily": "MaterialIcons",
          "fontSize": "24px",
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    >
      
    </div>
  </div>
  <div
    className="css-text-76zvg2"
    dir="auto"
    style={
      Object {
        "color": "rgba(0,0,0,0.32)",
      }
    }
  >
    edit
  </div>
</div>
`;

exports[`IconButton enabled matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxFlex": 1,
      "WebkitBoxOrient": "horizontal",
      "WebkitFlexBasis": "0%",
      "WebkitFlexDirection": "row",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "alignItems": "center",
      "cursor": "pointer",
      "flexBasis": "0%",
      "flexDirection": "row",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexAlign": "center",
      "msFlexDirection": "row",
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxPack": "center",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,1.00)",
        "borderBottomLeftRadius": "28px",
        "borderBottomRightRadius": "28px",
        "borderTopLeftRadius": "28px",
        "borderTopRightRadius": "28px",
        "height": "52px",
        "justifyContent": "center",
        "marginBottom": "7px",
        "marginLeft": "7px",
        "marginRight": "7px",
        "marginTop": "7px",
        "msFlexAlign": "center",
        "msFlexPack": "center",
        "width": "52px",
      }
    }
  >
    <div
      className="css-text-76zvg2"
      dir="auto"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.00)",
          "color": "rgba(40,44,52,1.00)",
          "fontFamily": "MaterialIcons",
          "fontSize": "24px",
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    >
      
    </div>
  </div>
  <div
    className="css-text-76zvg2"
    dir="auto"
    style={
      Object {
        "color": "inherit",
      }
    }
  >
    edit
  </div>
</div>
`;
